internal/server/data/group.go

```diff
@@ -71,59 +93,49 @@ func groupIDsForUser(tx ReadTxn, userID uid.ID) ([]uid.ID, error) {
 	return result, rows.Err()
 }
 
-func DeleteGroups(db GormTxn, selectors ...SelectorFunc) error {
-	toDelete, err := ListGroups(db, nil, selectors...)
+func DeleteGroup(tx WriteTxn, id uid.ID) error {
+	err := DeleteGrants(tx, DeleteGrantsOptions{BySubject: uid.NewGroupPolymorphicID(id)})
 	if err != nil {
-		return err
+		return fmt.Errorf("remove grants: %w", err)
 	}
 
-	ids := make([]uid.ID, 0)
-	for _, g := range toDelete {
-		ids = append(ids, g.ID)
-
-		err := DeleteGrants(db, DeleteGrantsOptions{BySubject: g.PolyID()})
-		if err != nil {
-			return err
-		}
-
-		identities, err := ListIdentities(db, nil, []SelectorFunc{ByOptionalIdentityGroupID(g.ID)}...)
-		if err != nil {
-			return err
-		}
-
-		var uidsToRemove []uid.ID
-		for _, id := range identities {
-			uidsToRemove = append(uidsToRemove, id.ID)
-		}
-		err = RemoveUsersFromGroup(db, g.ID, uidsToRemove)
-		if err != nil {
-			return err
-		}
+	_, err = tx.Exec(`DELETE from identities_groups WHERE group_id = ?`, id)
+	if err != nil {
+		return fmt.Errorf("remove useres from group: %w", err)
 	}
 
-	return deleteAll[models.Group](db, ByIDs(ids))
+	stmt := `
+		UPDATE groups
+		SET deleted_at = ?
+		WHERE id = ?
+		AND deleted_at is null
+		AND organization_id = ?`
+	_, err = tx.Exec(stmt, time.Now(), id, tx.OrganizationID())
+	return handleError(err)
 }
 
-func AddUsersToGroup(db GormTxn, groupID uid.ID, idsToAdd []uid.ID) error {
-	for _, id := range idsToAdd {
-		// This is effectively an "INSERT OR IGNORE" or "INSERT ... ON CONFLICT ... DO NOTHING" statement which
-		// works across both sqlite and postgres
-		_, err := db.Exec("INSERT INTO identities_groups (group_id, identity_id) SELECT ?, ? WHERE NOT EXISTS (SELECT 1 FROM identities_groups WHERE group_id = ? AND identity_id = ?)", groupID, id, groupID, id)
-		if err != nil {
-			return err
+func AddUsersToGroup(tx WriteTxn, groupID uid.ID, idsToAdd []uid.ID) error {
```

Comment:
This changed from a for loop performing one query per userID, to a single query that inserts multiple entries. It also uses `ON CONFLICT DO NOTHING` now that we only need to support postgreSQL.